### PR TITLE
Allow to disable compression in Apache HttpClient

### PR DESCRIPTION
**Pull request by tjuchniewicz**
Fixes gh-3122

### CODE REVIEW

File: spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonProperties.java [hunk 1]
```diff
@@ -22,7 +22,11 @@
 import static com.netflix.client.config.DefaultClientConfigImpl.DEFAULT_READ_TIMEOUT;
 
 public class RibbonProperties {
-    private final IClientConfig config;
+	private final IClientConfig config;
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonProperties.java]: indentation is off

Comment (tjuchniewicz) [reply spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonProperties.java]: Fixed. RibbonProperties use different indentation than other files.

**Comment by tjuchniewicz** [comment]
I will prepare test.
Regarding documentation. I've implemented `CommonClientConfigKey.GZipPayload'  from `ribbon-core` so we should find documentation there... but I can't see description of this config key.

**Comment by tjuchniewicz** [comment]
Unit tests added.

**Comment by ryanjbaxter** [comment]
Should we also make a similar change in `RibbonLoadBalancingHttpClient` in the case that Spring Retry is not on the classpath?

**Comment by tjuchniewicz** [comment]
`RibbonLoadBalancingHttpClient` is already updated.

**Comment by spencergibb** [comment]
Squashed and merged with polish via 0f7c64325e009941b49d5b0136ae6b72efe8a184

**Comment by tjuchniewicz** [comment]
@spencergibb Thank you.

**Pull request closed without merging**